distributed/scheduler.py

```diff
@@ -4138,6 +4193,52 @@ def decide_worker(self, ts):
 
         return worker
 
+    def transition_waiting_speculative(self, key):
+        try:
+
+            ts = self.tasks[key]
+
+            if self.validate:
+                # All dependencies are on the same worker
+                assert len({dts.processing_on for dts in ts.dependencies}) == 1
+
+            ws = list(ts.dependencies)[0].processing_on
+
+            worker = ws.address
+
+            duration = self.get_task_duration(ts)
+            # There are no comm costs if this is speculative
+            # comm = self.get_comm_cost(ts, ws)
+
+            ws.processing[ts] = duration
+            ts.processing_on = ws
+            ws.occupancy += duration
+            self.total_occupancy += duration
+            ts.state = "processing"
+            self.consume_resources(ts, ws)
+            self.check_idle_saturated(ws)
+            self.n_tasks += 1
+
+            if ts.actor:
+                ws.actors.add(ts)
+
+            ts.speculative = True
+
+            # logger.debug("Send job to worker: %s, %s", worker, key)
+
+            self.send_task_to_worker(worker, key)
+
+            return _recommend_speculative_assignment(ts)
```

Comment:
Ah, so we would send long chains down to a worker all at once?

Reply:
that's what currently happens and I can see it leading to less than stellar performance, especially if a cluster is scaling up -- 
We could add a config item about # of speculative tasks to assign to a given worker?  a la `max_height` in the fuse options?

Reply:
Oh, I think it's fine.  Let's leave it like this until there is a problem.